### DISCUSSION TITLE
File upload - url or client configuration

### DISCUSSION THREAD

**Discussion by tot-ra** [urql-graphql/urql#2905] [Q&A] [answered]
## Context
I've seen how [uploads are supposed to work](https://codesandbox.io/s/bjoernravenext-urql-multipart-rc1db) and read [the docs](https://formidable.com/open-source/urql/docs/advanced/persistence-and-uploads/#file-uploads). 

## Problem
The problem is that I use apollo router on backend which doesn't support file proxying to federated service https://github.com/apollographql/router/issues/842 so end service basically gets just empty object if I follow standard approach

```
CombinedError: [GraphQL] Variable "$file" got invalid value {}; Upload value invalid.
```

## Expectations
So what I used to do before, is to have two instances of _apollo client_ and for uploads, I would have different target URL (uploading directly to file management service).

Now, with _urql_, I don't see **how I can configure url or client instance for a particular mutation**. 

Is that possible? Here I see its taking url from core -> https://github.com/urql-graphql/urql/blob/main/exchanges/multipart-fetch/src/multipartFetchExchange.ts#LL37C24-L37C24

**Comment by JoviDeCroock** [comment, marked as answer]
You can pass url as part of the operationContext of your mutation, this is documented in the api part of our docs

The second argument for the execute function https://formidable.com/open-source/urql/docs/api/urql/#usemutation

**Comment by tot-ra** [comment]
Thanks, that solved it, I couldn't figure it out from typescript definitions 😄
